internal/engine/execution/queries.go

```diff
@@ -263,13 +263,22 @@ func createSchema(ctx context.Context, tx sql.TxMaker, schema *types.Schema, txi
 	return sp.Commit(ctx)
 }
 
-// getSchemas returns all schemas in the kwil_schemas table
-func getSchemas(ctx context.Context, tx sql.Executor) ([]*types.Schema, error) {
+// getSchemas returns all schemas in the kwil_schemas table.
+// convertFunc converts bytes into a schema. If nil, it will simply unmarshal the bytes.
+func getSchemas(ctx context.Context, tx sql.Executor, convertFunc func([]byte) (*types.Schema, error)) ([]*types.Schema, error) {
 	res, err := tx.Execute(ctx, sqlListSchemaContent)
 	if err != nil {
 		return nil, err
 	}
 
+	if convertFunc == nil {
+		convertFunc = func(b []byte) (*types.Schema, error) {
```

Comment:
Related concept, not requesting any changes, just food for thought: https://github.com/jchappelow/kwil-db/commit/423cdc6becfbe6690a5e054a9b4405f206345365
At least within the framework of JSON schema content, there are some simple ways to support reading prior schema versions if the schema doc and struct include a flag.  I suppose the only format we want to support migrating directly from text is KF, but we also support specifying a schema with JSON from kwil-cli so this is an exposed document structure with no real end-user accessible conversion.

Do we want to ditch the JSON option?  `  -t, --type string   file type of the database definition file (kf or json) (default "kf")`

Reply:
I'm a bit confused, why would we need to ditch the json option in the CLI? It's just used to submit an already parsed schema. I don't see how it is related at all, other than that schemas parsed with v0.7 will not be readable by the CLI

Reply:
It seems equally valid to provide your schema as an end user to the cli in kf or json.  Is it?

Reply:
It is. They obtain the json by using the parser though. I don't think we have ever directed anyone to construct a schema via json, they simply get it from the IDE (which has an export parse to json) feature.